### PR TITLE
Ticket8 patch votes with review id for review table

### PR DESCRIPTION
**Pull request by margaritisarg**

### CODE REVIEW

File: models/reviews.model.js [hunk 1]
```diff
@@ -1,5 +1,33 @@
...
+        `
+            UPDATE reviews
+                SET votes = votes + $1
+            WHERE review_id = $2
+            RETURNING *;   
+        `;
+        return db.query(sql, [votes, reviewID])
+        .then((data) => {
+            if(data.rowCount === 0) {
+                return {status: 204, msg: "204 - No content found"};
```
Comment (Pipe-Smoking-Rabbit) [inline models/reviews.model.js]: Status code 204 relates to a _successful_ request that has _no content_ to respond with, if the row count is zero in this case I would imagine that is a result of the review_id not existing which would be a 404 - Not found

File: models/reviews.model.js [hunk 2]
```diff
@@ -1,5 +1,33 @@
...
+            RETURNING *;   
+        `;
+        return db.query(sql, [votes, reviewID])
+        .then((data) => {
+            if(data.rowCount === 0) {
+                return {status: 204, msg: "204 - No content found"};            
+            }else{
+                const reviewObj = { title: data.rows[0].title, votes: data.rows[0].votes }
+                return {status: "Updated successfully", rowCount: data.rowCount, review: reviewObj}        
+            }
```
Comment (Pipe-Smoking-Rabbit) [inline models/reviews.model.js:23]: I think I see what you're doing here in trying to organise the response data into an object with info-keys before you send it back. Personally, I view this as taking away from the readability of your code.  My advice would be to keep it simple and just send back the result of the query: 
`return data.rows[0]`

Comment (Pipe-Smoking-Rabbit) [reply models/reviews.model.js:23]: You can then send back a simple response from the controller: 
`.then((review) => res.status(200).send({ review: review })`

File: __tests__/api.test.js [hunk 3]
```diff
@@ -169,6 +169,64 @@ describe("POST API - insert a comment row using review_id and author FK", () =>
...
+        return request(app)
+            .patch("/api/reviews/2")
+            .send(dummyData)
+            .expect(200)
+            .then(({body}) => {
+                const expected = {
+                    status: 'Updated successfully',
+                    rowCount: 1,
+                    review: { title: 'Jenga', votes: 55 }
+                  }
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/api.test.js]: Ideally you want to get back the whole review object with all it's keys

File: __tests__/api.test.js [hunk 4]
```diff
@@ -169,6 +169,64 @@ describe("POST API - insert a comment row using review_id and author FK", () =>
...
+        return request(app)
+            .patch("/api/reviews/2+")
+            .send(dummyData)
+            .expect(400)
+            .then(({body}) => {
+                expect(body).toEqual({msg: '400 - Bad input'})
+            })
+    })
+    test("400: bad input - incorrect ID in path char", () => {
+        const dummyData = {votes: 10}
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/api.test.js]: This test is possibly the same as the last test ( ID is not a number ), don't forget to also include a 404 test for if the ID is a number that doesn't exist in the database

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
Missing one test for if the review ID is a valid number but doesn't exist in the database. As a general comment I would recommend simplifying your responses, you're doing a lot currently and my worry is that it subtracts from the readability of your model function

**Review by Hard-Luck — APPROVED** [review]
Good to go! :)

**Pull request merged**